### PR TITLE
Add many links and reorganize libraries into sections

### PR DESCRIPTION
**Pull request by ehmicky**
This adds the following links:
  - [windows-build-tools](https://github.com/felixrieseberg/windows-build-tools) - install C++ Build Tools for Windows using npm.
  - [npm-windows-upgrade](https://github.com/felixrieseberg/npm-windows-upgrade) - Upgrade npm on Windows.
  - [MinGW](http://www.mingw.org/) - `gcc` on Windows.
  - [msys](http://www.mingw.org/wiki/msys) / [Git Bash](https://gitforwindows.org/) - Bash on Windows.
  - [node-windows](https://github.com/coreybutler/node-windows) - Windows support for Node.js scripts (daemons, eventlog, UAC, etc).
  - [Wine](https://www.winehq.org/): run Windows API calls on Linux, Mac, BSD and Solaris.
  - [Cygwin](https://www.cygwin.com/): run POSIX on Windows.
  - [WSL](https://docs.microsoft.com/en-us/windows/wsl/install-win10): run the Linux command line on Windows (ELF binary execution, system calls, filesystem, Bash, core utilities, common applications).
  - [node-notifier](https://github.com/mikaelbr/node-notifier) - Cross-platform desktop notifications.
  - [signal-exit](https://github.com/tapjs/signal-exit) - Cross-platform `exit` handler.
  - [readdirp](https://github.com/paulmillr/readdirp) - Recursive version of `fs.readdir()`.
  - [osenv](https://github.com/npm/osenv) - Cross-platform environment variables.
  - [username](https://github.com/sindresorhus/username) - Get the current username.
  - [getos](https://github.com/retrohacker/getos) - Retrieve current OS, including Linux distribution.
  - [is-wsl](https://github.com/sindresorhus/is-wsl) - Detect whether current platform is WSL (Windows Subsystem for Linux).
  - [os-name](https://github.com/sindresorhus/os-name) - Get the name of the current operating system.
  - [systeminformation](https://github.com/sebhildebrandt/systeminformation) - Hardware/software system information.
  - [is-elevated](https://github.com/sindresorhus/is-elevated) - Check if the process is running with elevated privileges.
  - [ps-list](https://github.com/sindresorhus/ps-list) - Get running processes.
  - [process-exists](https://github.com/sindresorhus/process-exists) - Check if a process exists.
  - [node-winreg](https://github.com/fresc81/node-winreg) - Access the Windows registry.
  - [rage-edit](https://github.com/MikeKovarik/rage-edit) - Access/modify the Windows registry.
  - [windows-registry-node](https://github.com/CatalystCode/windows-registry-node) - Access/modify the Windows registry and set file assocations.
  - link to core Node.js documentation, especially `os`, `path`, `fs`, `process` and `child_process`. Node.js and libuv already do lots of work to fix cross-platform issues, and knowing those core modules should be the first thing to learn, before diving into libraries IMHO.

This PR also reorganizes libraries into sections.

I have done this in a single PR because those would be too many individual PRs at once.

Please make sure:

  - [x] the title is clear and descriptive.
  - [x] you've read the [contribution guidelines](../blob/master/contributing.md).

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -41,34 +50,82 @@ Inspired by the [awesome](https://github.com/sindresorhus/awesome) list thing.
...
-- [cross-env](https://github.com/kentcdodds/cross-env) - Set environment variables cross-platform.
-- [cross-spawn](https://github.com/IndigoUnited/node-cross-spawn) - Cross-platform implementation of `child_process.spawn()`.
-- [clipboardy](https://github.com/sindresorhus/clipboardy) / [clipboard-cli](https://github.com/sindresorhus/clipboard-cli) - Cross-platform copy/paste.
+- [graceful-fs](https://github.com/isaacs/node-graceful-fs) - Improves the `fs` module, especially on Windows.
+- [any-path](https://github.com/bcoe/any-path) - Use Windows and POSIX paths interchangeably when fetching values from an object.
+
+### Signals
+
+- [fkill](https://github.com/sindresorhus/fkill-cli) - Kill processes. Cross-platform.
+- [signal-exit](https://github.com/tapjs/signal-exit) - Cross-platform `exit` handler.
```
Comment (bcoe) [inline readme.md]: wish this wasn't necessary; I wonder if it's worth considering adding similar functionality to Node.js itself.

Comment (ehmicky) [reply readme.md]: The issue is (as far as I know, correct me if I'm wrong) signals are a Unix concept. It's more of a convenience that Node.js and those libraries are translating some of the most common signals to  the way Windows does this (especially terminating processes).

Comment (sindresorhus) [reply readme.md]: @bcoe Definitely. `signal-exit` has also been a pain as it has caused a lot of subtle issues through the years. It's silly that Node.js doesn't have a hook that is guaranteed to be called on exit.

For example: https://github.com/sindresorhus/ora/pull/80 But also other issues I can't find right now.

Comment (sindresorhus) [reply readme.md]: > For me, if there was one thing I would love to see native support in Node.js, that's file permissions. At the moment, core Node.js only handles few cases based on file attributes (not Windows ACLs). The difference between Windows and Unix makes it impossible to work with them in a cross-platform. There are virtually 0 libraries on GitHub handling this. 

@MarkTiedemann Module idea ;)

Comment (MarkTiedemann) [reply readme.md]: @sindresorhus I'm still working on my "perfect" setup for creating Windows Node native addon modules (as a side project besides my day job, so progress is kinda slow). I'm going for a monorepo with many tiny modules in there.

But yeah, adding this module to the todo list. :) Thanks for the idea!

Comment (ehmicky) [reply readme.md]: @MarkTiedemann would love to see this happening... As I said I tried to get a hit at it, but I got frustrated at the complexity of [Windows permissions](https://docs.microsoft.com/en-us/windows/desktop/SecAuthZ/access-control). It uses three different systems: file attributes (readonly/hidden/etc.), ACLs/NTFS permissions (which are plentiful) and share permissions. User identification is using SIDs which are also different from how Unix does things. It's quite tricky to map Unix permissions to Windows permissions.

Also currently there are not lots of *quality* Node.js modules related to doing low-level Windows operations, such as manipulating the registry and modifying file attributes. There are few but they are not well maintained.

Comment (MarkTiedemann) [reply readme.md]: > There are few but they are not well maintained.

@ehmicky Right. That's what I wanna change.

**Pull request merged**